### PR TITLE
feat: migrate Consul

### PR DESCRIPTION
**Pull request by dstaley**
## "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [ ] The Asana task has been added to this PR's description
- [ ] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [ ] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [ ] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-git-dsconsul-migrate-hashicorp.vercel.app/) 🔎
- [Asana task](url) 🎟️

## What

Migrates Consul to dev-portal :tada:

## Testing

- [ ] Navigate to the preview: https://dev-portal-git-dsconsul-migrate-hashicorp.vercel.app/
- [ ] Select Consul from the product switcher in the bottom left
- [ ] Validate that the site matches https://www.consul.io/
- [ ] Validate that the docs are rendering as expected by clicking "Docs" in the menu
- [ ] Validate that the use case pages are rendering as expecting by clicking "Use Cases" in the menu and selecting a use case

### CODE REVIEW

File: scripts/migrate-io/migrate-consul.js [hunk 1]
```diff
@@ -10,18 +10,380 @@ const {
...
+      return true
+    })
+  )
+  //
+  // ASSETS
+  //
+  const assetsToCopy = [
+    // meta images
+    // '/img/favicons/favicon.ico', NOTE: actually a Consul favicon
+    // Favicon has been manually added as _consul-favicon.ico in public folder
```
Comment (BRKalow) [inline scripts/migrate-io/migrate-consul.js:144]: I'm not seeing a favicon for consul on this branch. (Not seeing one for Vault either 👀)

File: src/pages/_proxied-dot-io/consul/use-cases/[slug].tsx [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+  }
+
+  return {
+    props: {
+      data: page,
+    },
+    revalidate:
+      process.env.HASHI_ENV === 'production'
+        ? process.env.GLOBAL_REVALIDATE
+        : 10,
```
Comment (BRKalow) [inline src/pages/_proxied-dot-io/consul/use-cases/[slug].tsx:237]: Do we have `GLOBAL_REVALIDATE` defined anywhere?  💭

**Review by BRKalow — COMMENTED** [review]
Looks like we might be missing the favicon, otherwise this looks great! 🙏

File: src/pages/_proxied-dot-io/consul/use-cases/[slug].tsx (hunk 2 above)
Comment (dstaley) [reply src/pages/_proxied-dot-io/consul/use-cases/[slug].tsx:237]: Ah sorry, I forgot Vault and Consul used this!

File: scripts/migrate-io/migrate-consul.js (hunk 1 above)
Comment (dstaley) [reply scripts/migrate-io/migrate-consul.js:144]: I'll fix the Vault one in a separate PR!

**Review by BRKalow — APPROVED** [review]
Woohoooo

**Pull request merged**